sdks/go/pkg/beam/core/runtime/graphx/serialize.go

```diff
@@ -207,45 +207,45 @@ func encodeFn(u *graph.Fn) (*v1.Fn, error) {
 	case u.Fn != nil:
 		fn, err := encodeUserFn(u.Fn)
 		if err != nil {
-			return nil, fmt.Errorf("bad userfn: %v", err)
+			return nil, fmt.Errorf("failed to encode function %v, bad userfn: %v", u, err)
 		}
 		return &v1.Fn{Fn: fn}, nil
 
 	case u.Recv != nil:
 		t := reflect.TypeOf(u.Recv)
 		k, ok := runtime.TypeKey(reflectx.SkipPtr(t))
 		if !ok {
-			return nil, fmt.Errorf("bad recv: %v", u.Recv)
+			return nil, fmt.Errorf("failed to encode function %v, bad receiver %v", u, u.Recv)
```

Comment:
This one is trickier:
("failed to encode structural DoFn %v, bad receiver type: %T", u, u.Recv)

Reply:
Done, with a little more detail:

("failed to encode structural DoFn %v, failed to create TypeKey for receiver type %T", u, u.Recv)